src/Rings.jl

```diff
@@ -241,3 +241,49 @@ function is_finite(R::NCRing)
   c == 1 && return true
   throw(NotImplementedError(:is_finite, R))
 end
+
+@doc raw"""
+    is_noetherian(R::Ring)
+
+Check if the ring $R$ is Noetherian.
+```jldoctest
+julia> R, x = polynomial_ring(ZZ, [:x]);
+
+julia> is_noetherian(R)
+true
+```
+"""
+function is_noetherian(R::Ring)
+  throw(NotImplementedError(:is_noetherian, R))
+end
+
+is_noetherian(::Field) = true
+is_noetherian(::Integers) = true
+
+is_noetherian(R::Union{PolyRing, MPolyRing, LaurentPolyRing, LaurentMPolyRing}) = is_noetherian(coefficient_ring(R))
+is_noetherian(R::Union{MSeriesRing, SeriesRing}) = is_noetherian(base_ring(R))
+
+@doc raw"""
+    krull_dim(R::T) where T<:Ring
+
+Return the Krull dimension of the ring $R$. The method is currently only supported for certain commutative Noetherian Rings.
+```jldoctest
+julia> R, x = polynomial_ring(ZZ, [:x]);
+
+julia> krull_dim(R)
+2
+```
+"""
+function krull_dim(R::T) where T<:Ring
+  @req is_noetherian(R) "Krull dimension is only supported for Noetherian rings."
+  if R isa Integers
+    return 1
+  elseif R isa T where T<:Field
+    return 0
+  else
+    throw(NotImplementedError(:krull_dim, R))
+  end
+end
```

Comment:
This should be split up, like `is_noetherian`. Maybe like so:
```suggestion
function krull_dim end

krull_dim(::Field) = 0
krull_dim(::Integers) = 1

function krull_dim(R::T) where T<:Ring
  @req is_noetherian(R) "Krull dimension is only supported for Noetherian rings."
  throw(NotImplementedError(:krull_dim, R))
end
```
Of the final method then is again up for debate, because it doesn't really do much, and if in doubt, the call to `is_noetherian` itself may also not be implemented...